tokens/batched/token_request.go

```diff
@@ -38,44 +37,32 @@ func (r BatchedTokenRequest) Marshal() []byte {
 }
 
 func (r *BatchedTokenRequest) Unmarshal(data []byte) bool {
-	s := cryptobyte.String(data)
-
 	// At most, a quic varint is 4 byte long. copy them to read the length
 	if len(data) < 4 {
 		return false
 	}
 
 	l, offset := quicwire.ConsumeVarint(data)
-	s.Skip(offset)
 
 	r.token_requests = make([]tokens.TokenRequestWithDetails, 0)
-	i := 0
-	for i < int(l) {
-		var token_request_length uint16
-		if !s.ReadUint16(&token_request_length) {
-			return false
-		}
-		var token_request_data []byte
-		if !s.ReadBytes(&token_request_data, int(token_request_length)) {
-			return false
-		}
-		i += 2 + len(token_request_data)
+	i := offset
+	for i < offset+int(l) {
 		var token_request tokens.TokenRequestWithDetails
-		token_type := binary.BigEndian.Uint16(token_request_data[:2])
+		token_type := binary.BigEndian.Uint16(data[i:2])
 		switch token_type {
 		case type1.BasicPrivateTokenType:
 			token_request = new(type1.BasicPrivateTokenRequest)
 		case type2.BasicPublicTokenType:
 			token_request = new(type2.BasicPublicTokenRequest)
-		case type5.BatchedPrivateTokenType:
-			token_request = new(type5.BatchedPrivateTokenRequest)
 		default:
 			return false
 		}
-		if !token_request.Unmarshal(token_request_data) {
+		if !token_request.Unmarshal(data[i:]) {
 			return false
 		}
 		r.token_requests = append(r.token_requests, token_request)
+		// Super inneficient but we don't know how many bytes have benen read otherwise
+		i += len(token_request.Marshal())
```

Comment:
Could we set `r.raw` within the `Unmarshal` function (e.g., https://github.com/cloudflare/pat-go/blob/d2c46150eb0caa47691b59a790c088a2f04f95a9/tokens/type2/token_request.go#L47) so this marshaling is already cached?

Reply:
probably, but that would be specific to each token request implementation

Reply:
better to open an issue to fix Unmarshal later.